stan/math/prim/fun/matrix_exp_action_handler.hpp

```diff
@@ -104,26 +169,68 @@ class matrix_exp_action_handler {
    * @param [out] m int parameter m
    * @param [out] s int parameter s
    */
-  inline void set_approximation_parameter(const Eigen::MatrixXd& mat,
-                                          const double& t, int& m, int& s) {
-    if (l1norm(mat) < tol || t < tol) {
+  template <typename EigMat1, typename EigMat2,
+            require_all_eigen_t<EigMat1, EigMat2>* = nullptr,
+            require_all_st_same<double, EigMat1, EigMat2>* = nullptr>
+  inline void set_approx_order(const EigMat1& mat, const EigMat2& b,
+                               const double& t, int& m, int& s) {
+    // L1 norm
+    double normA = mat.colwise().template lpNorm<1>().maxCoeff();
+
+    if (normA < tol || t < tol) {
       m = 0;
       s = 1;
```

Comment:
I'd want to eagerly return here so it's clear that there's no more that happens below it. 

That will unnest the lower `else` statement and it'll be slightly easier to read.